app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,48 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def cross_link_amazon_node(new_result)
+      _, az, uid = URI(new_result[:identity_infra]).path.split('/')
+      az_id = AvailabilityZone.find_by(:name => az).try(:id)
+      Vm.where(:uid_ems              => uid,
+               :availability_zone_id => az_id,
+               :type                 => ManageIQ::Providers::Amazon::CloudManager::Vm.name)
+    end
+
+    def cross_link_google_node(new_result)
+      # TODO
+    end
+
+    def cross_link_node(new_result)
+      # Establish a relationship between this node and the vm it is on (if it is in the system)
+      # supported relationships: oVirt, Openstack and VMware.
+      uuid_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                    ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                    ManageIQ::Providers::Vmware::InfraManager::Vm.name]
+
+      # Searching for the underlying instance for Openstack or oVirt.
+      vms = Vm.where(:uid_ems => new_result[:identity_system].downcase, :type => uuid_types)
+      if vms.empty? && new_result[:identity_infra]
+        case URI(new_result[:identity_infra]).scheme
+        when "aws"
+          vms = cross_link_amazon_node(new_result)
+        when "gce"
+          vms = cross_link_google_node(new_result)
+        end
+      end
+
+      if vms.to_a.size == 1
```

Comment:
> We are comparing by an ID that is suppose to be unique so it should be more than one. Maybe we should change 'where' to 'find_by' on line 151 and in cross_link_amazon_node / cross_link_google_node ?

you probably meant "should NOT be more than one" ?
it's worth revisiting because all cross_link_X methods return value is assigned to vms in plural, and then there's a check if size == 1, so it's slightly misleading (but not that big of a deal)